db/dcrpg/pgblockchain.go

```diff
@@ -5399,16 +5409,44 @@ func (pgb *ChainDB) GetExplorerTx(txid string) *exptypes.TxInfo {
 		log.Errorf("Cannot create MsgTx for tx %v: %v", txhash, err)
 		return nil
 	}
+
+	var isMix bool
+	var mixDenom int64
+	var mixCount uint32
+	if !txhelpers.IsStakeTx(msgTx) {
+		isMix, mixDenom, mixCount = txhelpers.IsMixTx(msgTx)
+		if !isMix {
+			var ticketPrice int64
+			if txraw.Confirmations > 0 {
+				ticketPrice = pgb.GetSBitsByHash(txraw.BlockHash)
+			} else {
+				sdiffRes, err := pgb.Client.GetStakeDifficulty()
```

Comment:
Seems racy

Reply:
It is.  The tx may be confirmed and the sdiff changed by the time this is called.  I indent to address this.